src/app/reports/reports.component.ts

```diff
@@ -0,0 +1,107 @@
+/** Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import {MatPaginator, MatSort, MatTableDataSource} from '@angular/material';
+import { ActivatedRoute, Router } from '@angular/router';
+
+/**
+ * Reports component.
+ */
+@Component({
+  selector: 'mifosx-app-reports',
+  templateUrl: './reports.component.html',
+  styleUrls: ['./reports.component.scss']
+})
+export class ReportsComponent implements OnInit {
+
+  /** Reports data. */
+  reportsData: any;
+  /** Report category filter. */
+  filter: string;
+  /** Columns to be displayed in reports table. */
+  displayedColumns: string[] = ['reportName', 'reportType', 'reportCategory'];
+  /** Data source for reports table. */
+  dataSource = new MatTableDataSource();
+
+  /** Paginator for reports table. */
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  /** Sorter for reports table. */
+  @ViewChild(MatSort) sort: MatSort;
+
+  /**
+   * Retrieves the reports data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router: Route.
+   */
+  constructor(private route: ActivatedRoute, private router: Router) {
+    this.router.routeReuseStrategy.shouldReuseRoute = () => false;
```

Comment:
Please explain the usage of this here.

Reply:
Hi Abhay it isn't working fine on my system if I remove this line, are you sure it's working fine in yours? 

![Screenshot from 2020-04-26 22-49-24](https://user-images.githubusercontent.com/54709463/80314807-4e6fe900-8811-11ea-9041-2655b070a7f7.png)

On savings page report category is clients, reproduce by navigating to clients reports and then to savings.

![Screenshot from 2020-04-26 23-02-32](https://user-images.githubusercontent.com/54709463/80314902-09988200-8812-11ea-99ff-7352336f4e93.png)

should re-use route returns true if current and future route configs match which they do because navigating b/w savings and clients `reports/:filter` config doesn't change, so angular re uses value of `filter` as 'client'

Reply:
Alright, yes it is not working correctly without this. Let's add a comment in the file here to explain its usage to ensure we remember it and then we can merge.

Reply:
updated